### PR TITLE
Fix deg and lev parsing; fix for deg=1; allow subvectors of size 1

### LINKED ISSUE #215: specifying level or degree alone causes crash
If either level or degree is specified as a command line option, and both are not specified, the program will crash. not great.

### PR DESCRIPTION
**Pull request by bmcdanie**
closes #215

### CODE REVIEW

File: src/tensors.hpp [hunk 1]
```diff
@@ -1306,7 +1306,7 @@ fk::vector<P, mem, resrc>::extract(int const start, int const stop) const
 {
   assert(start >= 0);
   assert(stop < this->size());
-  assert(stop > start);
+  assert(stop >= start);
```
Comment (grahamlopez) [inline src/tensors.hpp]: I'm not convinced that our constructors handle allocations of `-1` very well

File: src/pde/pde_base.hpp [hunk 2]
```diff
@@ -365,16 +365,20 @@ class PDE
     // if default lev/degree not used
     if (num_levels != -1 || degree != -1)
     {
```
Comment (grahamlopez) [inline src/pde/pde_base.hpp]: ah, so when you said that we crashed before, we were actually hitting `asserts()`? If so, that still sucks, but not nearly as troubling.

Comment (bmcdanie) [reply src/pde/pde_base.hpp]: yes

File: src/tensors.hpp (hunk 1 above)
Comment (bmcdanie) [reply src/tensors.hpp]: we do appear to be missing some asserts in some of the constructors, I can add those

Comment (bmcdanie) [reply src/tensors.hpp]: added

Comment (grahamlopez) [reply src/tensors.hpp]: I'm referring to line 1314

Comment (bmcdanie) [reply src/tensors.hpp]: I think the least possible value of `sub_size` there is one, unless I've made a mistake. it used to permit a minimum vector size of 2, but I changed that to allow for single element subvects.

Comment (grahamlopez) [reply src/tensors.hpp]: you are right; I'm doing too many things at once

**Comment by grahamlopez** [comment]
test this please

**Pull request merged**